frontend/src/types/NMPFileField.ts

```diff
@@ -118,4 +122,5 @@ export type NMPFileField = {
   fertigations: NMPFileFertigation[];
   otherNutrients: NMPFileOtherNutrient[];
   previousYearManureApplicationNCredit?: number;
+  soilNitrateCredit: NMPFileSoilNitrateCredit[];
```

Comment:
Why is this stored as an array? Is it possible to have more than one nitrate credit?

If no, I would recommend writing the type as `[NMPFileSoilNitrateCredit] || []` if you still want to store the row in an array.